desc/configuration.py

```diff
@@ -1303,78 +1254,35 @@ def body_fun(k_rhok_thetak_Rk_Zk):
+
         Parameters
         ----------
-        nsurfs : int, optional
-            Number of flux surfaces to check (Default = 10).
-        ntheta : int, optional
-            Number of straight field-line poloidal contours to check (Default = 20).
-        nzeta : int, optional
-            Number of toroidal planes to check. By default checks the zeta=0
-            plane for axisymmetric equilibria and 5 planes evenly spaced in
-            zeta between 0 and 2pi/NFP for non-axisymmetric equilibria.
-            Otherwise uses nzeta planes linearly spaced in zeta between 0 and 2pi/NFP.
-        Nt : int, optional
-            Number of theta coordinates to use for the rho contours (Default = 45).
-        Nr : int, optional
-            Number of rho coordinates to use for the theta contours (Default = 20).
+        grid  :  Grid, optional
+            Grid on which to evaluate the coordinate jacobian and check for the sign. (Default to QuadratureGrid with eq's current grid resolutions)
 
         Returns
         -------
         is_nested : bool
             whether or not the surfaces are nested
 
         """
-        planes_nested_bools = []
-        if nzeta is None:
-            zetas = (
-                [0.0]
-                if self.N == 0
-                else np.linspace(0, 2 * np.pi / self.NFP, 5, endpoint=False)
-            )
-        else:
-            zetas = np.linspace(0, 2 * np.pi / self.NFP, nzeta, endpoint=False)
-
-        for zeta in zetas:
-            r_grid = LinearGrid(rho=nsurfs, theta=Nt, zeta=zeta, endpoint=True)
-            t_grid = LinearGrid(rho=Nr, theta=ntheta, zeta=zeta, endpoint=False)
-
-            r_coords = self.compute("R", r_grid)
-            t_coords = self.compute("lambda", t_grid)
-
-            v_nodes = t_grid.nodes
-            v_nodes[:, 1] = t_grid.nodes[:, 1] - t_coords["lambda"]
-            v_grid = Grid(v_nodes)
-            v_coords = self.compute("R", v_grid)
-
-            # rho contours
-            Rr = r_coords["R"].reshape(
-                (r_grid.num_rho, r_grid.num_theta, r_grid.num_zeta)
-            )[:, :, 0]
-            Zr = r_coords["Z"].reshape(
-                (r_grid.num_rho, r_grid.num_theta, r_grid.num_zeta)
-            )[:, :, 0]
-
-            # theta contours
-            Rv = v_coords["R"].reshape(
-                (t_grid.num_rho, t_grid.num_theta, t_grid.num_zeta)
-            )[:, :, 0]
-            Zv = v_coords["Z"].reshape(
-                (t_grid.num_rho, t_grid.num_theta, t_grid.num_zeta)
-            )[:, :, 0]
-
-            rline = MultiLineString(
-                [LineString(np.array([R, Z]).T) for R, Z in zip(Rr, Zr)]
-            )
-            vline = MultiLineString(
-                [LineString(np.array([R, Z]).T) for R, Z in zip(Rv.T, Zv.T)]
-            )
+        if R_lmn is None:
+            R_lmn = self.R_lmn
+        if Z_lmn is None:
+            Z_lmn = self.Z_lmn
+
+        data = self.compute(name="sqrt(g)", grid=grid)
```

Comment:
might need to call compute_jacobian directly here in order to pass in R_lmn,Z_lmn

Reply:
or remove R,Z as args since we're not using it as a callback inside the optimizer anymore

Reply:
you're right, the way I do it now does not even use the given R_lmn, Z_lmn, oops

Reply:
I'll retain the R_lmn, Z_lmn in case it could be useful to use this to assess nestedness without making a configuration's R_lmn, Z_lmn different